android/app/src/main/java/app/organicmaps/editor/OsmLoginBottomFragment.java

```diff
@@ -0,0 +1,115 @@
+import android.widget.ProgressBar;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.android.material.bottomsheet.BottomSheetBehavior;
+import com.google.android.material.bottomsheet.BottomSheetDialog;
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
+import com.google.android.material.textfield.TextInputEditText;
+
+import app.organicmaps.R;
+import app.organicmaps.util.Constants;
+import app.organicmaps.util.InputUtils;
+import app.organicmaps.util.UiUtils;
+import app.organicmaps.util.Utils;
+import app.organicmaps.util.concurrency.ThreadPool;
+import app.organicmaps.util.concurrency.UiThread;
+
+public class OsmLoginBottomFragment extends BottomSheetDialogFragment {
+    final private OsmLoginFragment parentFragment;
+    private TextInputEditText mLoginInput;
+    private TextInputEditText mPasswordInput;
+    private Button mLoginButton;
+    private Button mLostPasswordButton;
+    private Button mRegisterButton;
+    private ProgressBar mProgress;
+
+    public OsmLoginBottomFragment(OsmLoginFragment parentFragment) {
+        super();
+        this.parentFragment = parentFragment;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_osm_login_bottom, container, false);
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        BottomSheetDialog dialog = (BottomSheetDialog) super.onCreateDialog(savedInstanceState);
+        dialog.getBehavior().setState(BottomSheetBehavior.STATE_EXPANDED);
+        return dialog;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        mLoginInput = view.findViewById(R.id.osm_username);
+        mPasswordInput = view.findViewById(R.id.osm_password);
+
+        mLoginButton = view.findViewById(R.id.login);
+        mLoginButton.setOnClickListener((v) -> doLogin());
+
+        mLostPasswordButton = view.findViewById(R.id.lost_password);
+        mLostPasswordButton.setOnClickListener((v) -> Utils.openUrl(requireActivity(), Constants.Url.OSM_RECOVER_PASSWORD));
+        mRegisterButton = view.findViewById(R.id.register);
+        mRegisterButton.setOnClickListener((v) -> Utils.openUrl(requireActivity(), Constants.Url.OSM_REGISTER));
+        mProgress = view.findViewById(R.id.osm_login_progress);
+    }
+
+    private void doLogin() {
+        InputUtils.hideKeyboard(mLoginInput);
+        final String username = mLoginInput.getText().toString().trim();
+        final String password = mPasswordInput.getText().toString();
+        enableInput(false);
+        UiUtils.show(mProgress);
+        mLoginButton.setText("");
```

Comment:
Why the text is cleared?

Reply:
Layout has the button and the progress bar at the same place. When authentication is in progress we disable button and show progress bur. If we don't clear button text it would look like:

![vlcsnap-2024-08-20-09h34m13s156](https://github.com/user-attachments/assets/af1ec057-3c19-4eb4-b925-a64998f34066)